BEST_PRACTICES.md

```diff
@@ -0,0 +1,199 @@
+# MCP Server Best Practices
```

Comment:
nit: should we just go ahead and create a `docs` directory now? or keep this at the top-level because it's a single doc? Up to you.